File: fibonacci_recursion.py
#!/usr/bin/env python3
# Выводит n-й член ряда Фибоначчи
# и всю последовательность до n
from time import time, clock

fib_arr = []


def fibonacci(n):
    if n < 2:
        return(n)
    else:
        return(fibonacci(n - 2) + fibonacci(n - 1))


n = int(input('''
    Введите порядковый номер числа
    из последовательности Фибоначчи:'''))

start_time = time()
start_clock = clock()
for i in range(n + 1):
    fib_arr.append(fibonacci(i))
stop_time = time()
stop_clock = clock()

print("%d-ый член последовательности Фибоначчи - это: %d" % (n, fib_arr[n]))
print("Вся последовальеность до %d-го члена:" % (n))
print(fib_arr)
print("Время вычисления по time: %s секунд" % (stop_time - start_time))
print("Время вычисления по clock: %s секунд" % (stop_clock - start_clock))

File: fibonacci_recursion.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"time"
)

var t0, t1 time.Time
var fib_arr = []int{}

func fibonacci(n int) int {
	if n < 2 {
		return n
	} else {
		return (fibonacci(n-2) + fibonacci(n-1))
	}
}

func main() {
	if len(os.Args) < 2 {
		fmt.Println(`
			Введите порядковый номер числа
			из последовательности Фибоначчи
			как аргумент запуска приложения.
			`)
	} else {
		n, err := strconv.Atoi(os.Args[1])
		if err != nil {
			fmt.Println(err)
			os.Exit(2)
		}
		t0 = time.Now()
		for i := 0; i <= n; i++ {
			fib_arr = append(fib_arr, fibonacci(i))
		}
		t1 = time.Now()
		fmt.Printf("%v-й член последовательности Фибоначчи - это: %v\n", n, fib_arr[n])
		fmt.Printf("Вся последовальеность до %v-го члена:\n", n)
		fmt.Println(fib_arr)
		fmt.Printf("Время вычисления составило %v.\n", t1.Sub(t0))
	}

}
